sr_description/hand/xacro/th_ff_rf_ellipsoid_hand.urdf.xacro

```diff
@@ -18,59 +9,14 @@ xmlns:xacro="http://www.ros.org/wiki/xacro">
   <xacro:macro name="shadowhand"
   params="muscletrans muscle bio bt_sp ubi eli reflect prefix">
-    <!-- Forearm -->
-    <xacro:forearm muscle="${muscle}" prefix="${prefix}" />
-    <!-- Wrist -->
-    <xacro:if value="${muscle}">
-      <xacro:wrist muscletrans="${muscletrans}" prefix="${prefix}"
-      reflect="${reflect}" parent="${prefix}forearm">
-        <origin xyz="0 0 0.4096" rpy="0 0 0" />
-      </xacro:wrist>
-    </xacro:if>
-    <xacro:unless value="${muscle}">
-      <xacro:wrist muscletrans="false" prefix="${prefix}"
-      reflect="${reflect}" parent="${prefix}forearm">
-        <origin xyz="0 -0.010 0.213" rpy="0 0 0" />
-      </xacro:wrist>
-    </xacro:unless>
-    <!-- Palm -->
-    <xacro:palm muscletrans="${muscletrans}" prefix="${prefix}"
-    reflect="${reflect}" parent="${prefix}wrist" />
-    <!-- First Finger Biotac -->
-    <xacro:finger muscletrans="${muscletrans}" lf="false"
-    bio="false" bt_sp="false" ubi="false" eli="true" prefix="${prefix}"
-    reflect="${reflect}" link_prefix="ff" joint_prefix="FF"
-    parent="${prefix}palm">
-      <origin xyz="${reflect*0.033} 0 0.095" rpy="0 0 0" />
-      <axis xyz="0 ${-1*reflect} 0" />
-    </xacro:finger>
-    <!-- Middle Finger -->
-    <xacro:finger muscletrans="${muscletrans}" lf="false"
-    bio="${bio}" bt_sp="${bt_sp}" ubi="${ubi}" eli="${eli}" prefix="${prefix}"
-    reflect="${reflect}" link_prefix="mf" joint_prefix="MF"
-    parent="${prefix}palm">
-      <origin xyz="${reflect*0.011} 0 0.099" rpy="0 0 0" />
-      <axis xyz="0 ${-1*reflect} 0" />
-    </xacro:finger>
-    <!-- Ring Finger -->
-    <xacro:finger muscletrans="${muscletrans}" lf="false"
-    bio="false" bt_sp="false" ubi="false" eli="true" prefix="${prefix}"
-    reflect="${reflect}" link_prefix="rf" joint_prefix="RF"
-    parent="${prefix}palm">
-      <origin xyz="${reflect*-0.011} 0 0.095" rpy="0 0 0" />
-      <axis xyz="0 ${reflect} 0" />
-    </xacro:finger>
-    <!-- Little Finger  -->
-    <xacro:finger muscletrans="${muscletrans}" lf="true"
-    bio="${bio}" bt_sp="${bt_sp}" ubi="${ubi}" eli="${eli}" prefix="${prefix}"
-    reflect="${reflect}" link_prefix="lf" joint_prefix="LF"
-    parent="${prefix}palm">
-      <origin xyz="0 0 0.06579" rpy="0 0 0" />
-      <axis xyz="0 ${reflect} 0" />
-    </xacro:finger>
-    <!-- Thumb Biotac -->
-    <xacro:thumb muscletrans="${muscletrans}" bio="false" bt_sp="false"
-    ubi="false" eli="true" prefix="${prefix}" reflect="${reflect}"
-    is_lite="0" parent="${prefix}palm"></xacro:thumb>
+    <!-- Ensure tip_sensors is defined before usage below -->
+    <xacro:property name="tip_sensors" default="${5 * ['std']}" />
+    <!-- translate old parameters to new ones, eventually updating tip_sensors -->
+    <xacro:include filename="$(find sr_description)/hand/xacro/backward_compat.urdf.xacro" />
+    <!-- forward to generic macro -->
+    <xacro:include filename="$(find sr_description)/hand/xacro/full_hand.urdf.xacro" />
```

Comment:
If I am not mistaken, full_hand.urdf.xacro already includes the backward compat. Isn't it weird to do 2 inclusion here ?

Additionally, those specific macros created in the past for specific hands that could not be instantiated with the generic macro, were deleted in #109 because they are only used by a single robots/*.xacro . there is no need to make these backward compatible IMHO. People out there who were including those very specific macros in their own file should upgrade (so it should break for them), and they are very few, as compared to the main full_hand.xacro macro that was used in far more hands, (so should not break for them). If they were using the robots/*.xacro then the files can use the new macro as in https://github.com/shadow-robot/sr_common/pull/109/commits/9ed52f1dced2a31b733d03f01e87146f4d0348cf#diff-3c1228053895eb1008febb3f202141b33d09e83fb896afb77aa9c6c4a6b48799

Reply:
We should not break existing code without a deprecation phase.

Reply:
I accept you statement on the deprecation phase. 
You did not answer on the 2 inclusions when another file also uses the "forward" but without the 2 inclusions

Reply:
> If I am not mistaken, full_hand.urdf.xacro already includes the backward compat. Isn't it weird to do 2 inclusion here ?

As we want to [access tip_sensors a few lines below](https://github.com/ubi-agni/sr_common/blob/23c55bc11d0512a7934185e2d2c1350999d52ea6/sr_description/hand/xacro/th_ff_rf_ellipsoid_hand.urdf.xacro#L20-L21), we need to ensure that it is well-defined from the old-style params *before* we call the macro, which then would include the compat file again.

Reply:
I understand the explanation now, that is just to provide the possibility to set the other fingers to any given sensors. I see now that such a case was possible with the macro before, but I had not anticipated this case since the robots/*.xacro matching this macro file uses 'std' for the 2 other fingers.